include/quic/connection.hpp

```diff
@@ -20,62 +20,111 @@ namespace oxen::quic
 {
     class Endpoint;
     class Stream;
-    class Server;
-    class Client;
-    class Handler;
 
-    class Connection
+    class connection_interface
     {
+      public:
+        virtual std::shared_ptr<Stream> get_new_stream(
+                stream_data_callback_t data_cb = nullptr, stream_close_callback_t close_cb = nullptr) = 0;
+    };
+
+    class Connection : public connection_interface, public std::enable_shared_from_this<Connection>
+    {
+      protected:
+        friend class Network;
+        friend class Endpoint;
+
+        static std::shared_ptr<Connection> make_inbound_conn(
+                Endpoint& ep,
+                const ConnectionID& scid,
+                const ConnectionID& dcid,
+                const Path& path,
+                std::shared_ptr<uv_udp_t> handle,
+                std::shared_ptr<TLSCreds> creds,
+                config_t u_config,
+                ngtcp2_pkt_hd* hdr)
+        {
+            log::trace(log_cat, "{} called", __PRETTY_FUNCTION__);
+            return _make_conn(ep, scid, dcid, path, handle, std::move(creds), u_config, Direction::INBOUND, hdr);
+        };
+
+        static std::shared_ptr<Connection> make_outbound_conn(
+                Endpoint& ep,
+                const ConnectionID& scid,
+                const ConnectionID& dcid,
+                const Path& path,
+                std::shared_ptr<uv_udp_t> handle,
+                std::shared_ptr<TLSCreds> creds,
+                config_t u_config)
+        {
+            log::trace(log_cat, "{} called", __PRETTY_FUNCTION__);
+            return _make_conn(ep, scid, dcid, path, handle, std::move(creds), u_config, Direction::OUTBOUND);
+        };
```

Comment:
These functions seems verbose; would it be easier to just expose `make_conn` as a public static method, since all these do is simply forward all their arguments (plus a Direction) to it?

Reply:
So currently `make_{inbound,outbound}_conn` is protected, but the `_make_conn` is private. There's definitely an argument for bypassing the inbound/outbound functions entirely, but what purpose would making `_make_conn` public serve? Currently it's only called by `Endpoint`, which has friend access

Reply:
Deleting `make_{inbound,outbound}_conn` and making `make_conn` private per this morning's discussion

Motivation: gating access and removing `friend` class specifiers in adding things to `connection_interface`